src/ansible_navigator/runner/base.py

```diff
@@ -0,0 +1,218 @@
+        self._eei = execution_environment_image
+        self._navigator_mode = navigator_mode
+        self._set_environment_variable: Dict[str, Any] = (
+            set_environment_variable if isinstance(set_environment_variable, dict) else {}
+        )
+        self._pass_environment_variable: List[str] = (
+            pass_environment_variable if isinstance(pass_environment_variable, list) else []
+        )
+        self._host_cwd = host_cwd
+        self._rotate_artifacts = rotate_artifacts if isinstance(rotate_artifacts, int) else None
+        self._timeout = timeout if isinstance(timeout, int) else None
+        self.ansible_runner_instance: Runner
+        self.cancelled: bool = False
+        self.finished: bool = False
+        self.status: Optional[str] = None
+        self._runner_args: Dict = {}
+        self._runner_artifact_dir: Optional[str] = None
+        if self._ee:
+            self._runner_args.update(
+                {
+                    "container_image": self._eei,
+                    "process_isolation_executable": self._ce,
+                    "process_isolation": True,
+                    "container_volume_mounts": container_volume_mounts,
+                    "container_options": container_options,
+                    "container_workdir": container_workdir,
+                }
+            )
+        self._runner_args.update(
+            {
+                "private_data_dir": self._private_data_dir,
+                "json_mode": True,
+                "quiet": True,
+                "cancel_callback": self.runner_cancelled_callback,
+                "finished_callback": self.runner_finished_callback,
+                "artifacts_handler": self.runner_artifacts_handler,
+                "timeout": self._timeout,
+            }
+        )
+        if self._rotate_artifacts is not None:
+            self._runner_args.update({"rotate_artifacts": self._rotate_artifacts})
+
+        self._add_env_vars_to_args()
+
+        if self._host_cwd:
+            # ensure the cwd ends with a trailing slash
+            host_cwd = os.path.join(self._host_cwd, "")
+            self._runner_args.update({"host_cwd": host_cwd})
+
+        if self._navigator_mode == "stdout":
+            self._runner_args.update(
+                {"input_fd": sys.stdin, "output_fd": sys.stdout, "error_fd": sys.stderr}
+            )
+
+    def __del__(self):
+        """
+        class destructor, handle runner artifact file deletion is rotate_artifacts \
+        value is not set
+        """
+        if (
+            self._rotate_artifacts is not None
+            and self._runner_artifact_dir
+            and os.path.exists(self._runner_artifact_dir)
+        ):
+
+            self._logger.debug(
+                "delete ansible-runner artifact directory at path %s", self._runner_artifact_dir
+            )
+            shutil.rmtree(self._runner_artifact_dir, ignore_errors=True)
+
+    @staticmethod
```

Comment:
Unless you expect this to be reused by subclassed, move it to the module-global level. In Python, it's an antipattern to use classes for namespacing pure functions.

Reply:
The use of this is limited to the class, hence the _ name and inclusion within the class itself.